### PR TITLE
Allows messages to be maps with string keys

### LINKED ISSUE #34: Error in BroadwayCloudPubSub.GoogleApiClient.wrap_received_messages/2
For some reason that I can't figure out, when querying Pub Sub in
production, line 71 in GoogleApiClient the pubsub_projects_subscriptions_pull/3
returns me something that looks like this:

```elixir
 {:ok,
 %GoogleApi.PubSub.V1.Model.PullResponse{
   receivedMessages: [
     %{
       "ackId" => "ID",
       "message" => %{
         "attributes" => %{"space" => "vegans"},
         "data" => "VEhJUyBJUyBBIFRFU1Q=",
         "messageId" => "1234321",
         "publishTime" => "2019-09-05T17:41:31.520Z"
       }
     }
   ]
 }}
```

However I notice when I run the tests here the same function returns me:

```elixir
{:ok,
 %GoogleApi.PubSub.V1.Model.PullResponse{
   receivedMessages: [
     %GoogleApi.PubSub.V1.Model.ReceivedMessage{
       ackId: "ACK ID",
       message: %GoogleApi.PubSub.V1.Model.PubsubMessage{
         attributes: %{"space" => "vegans"},
         data: "VEhJUyBJUyBBIFRFU1Q=",
         messageId: "708601755676243",
         publishTime: ~U[2019-09-05 17:41:31.520Z]
       }
     }
   ]
 }}
```

Any ideas?

### PR DESCRIPTION
**Pull request by Adzz**
For some reason that I can't figure out, when querying Pub Sub in
production, line 71 in GoogleApiClient the pubsub_projects_subscriptions_pull/3
returns me something that looks like this:

```elixir
 {:ok,
 %GoogleApi.PubSub.V1.Model.PullResponse{
   receivedMessages: [
     %{
       "ackId" => "ID",
       "message" => %{
         "attributes" => %{"space" => "vegans"},
         "data" => "VEhJUyBJUyBBIFRFU1Q=",
         "messageId" => "1234321",
         "publishTime" => "2019-09-05T17:41:31.520Z"
       }
     }
   ]
 }}
```

However I notice when I run the tests here the same function returns me:

```elixir
{:ok,
 %GoogleApi.PubSub.V1.Model.PullResponse{
   receivedMessages: [
     %GoogleApi.PubSub.V1.Model.ReceivedMessage{
       ackId: "ACK ID",
       message: %GoogleApi.PubSub.V1.Model.PubsubMessage{
         attributes: %{"space" => "vegans"},
         data: "VEhJUyBJUyBBIFRFU1Q=",
         messageId: "708601755676243",
         publishTime: ~U[2019-09-05 17:41:31.520Z]
       }
     }
   ]
 }}
```

This allows me to continue even though my response from Pub Sub is not
getting serialized into a GoogleApi.PubSub.V1.Model.ReceivedMessage

closes #34

### CODE REVIEW

**Comment by mcrumm** [comment]
I think this does not belong here, since the underlying Google library is ultimately responsible for (de)serialization.

Additionally, with this change a consumer can no longer be certain of the shape of the message data coming out of the producer, as they'll now have to match on both atom and string keys.

**Comment by josevalim** [comment]
Agreed. We should rather try to figure out exactly why you are getting a map of string keys. Are you running the latest? What is your broadway, broadway_cloud_pub_ssub and google API versions?

**Comment by Adzz** [comment]
Thanks all, I agree too, not proud of the code particularly just didn't know what else to do.

I updated the issue https://github.com/plataformatec/broadway_cloud_pub_sub/issues/34 , maybe lets continue the conversation there so we don't get confused! But am using latest versions of the libraries

File: mix.lock [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "earmark": {:hex, :earmark, "1.3.5", "0db71c8290b5bc81cb0101a2a507a76dca659513984d683119ee722828b424f6", [:mix], [], "hexpm"},
   "ex_doc": {:hex, :ex_doc, "0.21.1", "5ac36660846967cd869255f4426467a11672fec3d8db602c429425ce5b613b90", [:mix], [{:earmark, "~> 1.3", [hex: :earmark, repo: "hexpm", optional: false]}, {:makeup_elixir, "~> 0.14", [hex: :makeup_elixir, repo: "hexpm", optional: false]}], "hexpm"},
   "gen_stage": {:hex, :gen_stage, "0.14.2", "6a2a578a510c5bfca8a45e6b27552f613b41cf584b58210f017088d3d17d0b14", [:mix], [], "hexpm"},
-  "google_api_pub_sub": {:hex, :google_api_pub_sub, "0.11.0", "ed8d65d28ce7fb2bf05ffb97ba678ab81dc1e3fe696c34bf2ff20f3a385dd61f", [:mix], [{:google_gax, "~> 0.2", [hex: :google_gax, repo: "hexpm", optional: false]}], "hexpm"},
+  "google_api_pub_sub": {:hex, :google_api_pub_sub, "0.14.0", "6babac64ac8ee2e705a180ad2c1ac15a58e68cbeee506aaa2e96f55b6aaa06da", [:mix], [{:google_gax, "~> 0.2", [hex: :google_gax, repo: "hexpm", optional: false]}], "hexpm"},
```
Comment (wojtekmach) [inline mix.lock]: no problem in bumping this, but there doesn't seem to be any meaningful changes:

https://diff.jola.dev/diff/google_api_pub_sub/0.11.0/0.14.0

if there was a regression with regards to json handling, perhaps it was in google_gax or tesla deps?

Comment (mcrumm) [reply mix.lock]: @wojtekmach I've been looking, but I can't find anything significant.

**Pull request closed without merging**